### DISCUSSION TITLE
Multi body RAOs

### DISCUSSION THREAD

**Discussion by sdillenburg** [capytaine/capytaine#177] [Q&A] [answered]
Dear Matthieu,
Dear All,

I would like to compute motion RAOs for a body which is part of a multi-body system. Computing the hydrodynamic data is not the issue, but I am unsure how to calculate the RAOs correctly. Radiation of one body influences the incident wave on the other (and vice versa).

I had a look in my textbooks but could not find a description of how this is accomplished.

Do you have a hint where to find information on this in the literature? Any hint is appreciated.

Many thanks and best regards
Simon

**Comment by mancellin** [comment, marked as answer]
Hi Simon,

I don't have a precise reference in the litterature to give you right now.

In Capytaine, the standard way to treat this kind of problem is to merge the `N` bodies in a single `FloatingBody` with `6N` dofs. Then the off-diagonal terms of the `6N×6N` added mass and radiation damping matrices account for the interactions between bodies. The other matrices (inertia, hydrostatic stiffness) are block-diagonal `6N×6N` matrices (that is `N` blocks of size `6×6` along the diagonal) and the RAO is a `6N` vector obtained as usual.

Then if `N` is large, there are several ways to limit the computational cost by solving only approximately the interaction between distant bodies. One of them had been discussed in #51.

**Reply by sdillenburg** [reply]
Thanks for your reply, which helps a lot. I will give it a try and let you know what the outcome is ;)
